sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -117,7 +117,12 @@ EventImplPtr Scheduler::addCG(std::unique_ptr<detail::CG> CommandGroup,
     bool Enqueued;
 
     auto CleanUp = [&]() {
-      if (NewCmd && (NewCmd->MDeps.size() == 0 && NewCmd->MUsers.size() == 0)) {
+      if (NewCmd && (NewCmd->MDeps.size() == 0 && NewCmd->MUsers.size() == 0 &&
+                     NewCmd->getPreparedHostDepsEvents().size() == 0)) {
+        // Calling assert to prevent cleaning up of the command which enqueue
+        // status is not successfull
+        assert(NewCmd->isSuccessfullyEnqueued());
```

Comment:
If this is a requirement for the cleanup to do anything, does it make sense to call it from the fail-cases? Most often they will report `EnqueueResultT::SyclEnqueueFailed` and thus not report the command as having been enqueued succesfully.